prusti/src/ide_helper/query_signature.rs

```diff
@@ -0,0 +1,432 @@
+use prusti_common::config;
+use std::{collections::HashMap, fmt};
+
+use super::compiler_info::ProcDef;
+use prusti_rustc_interface::{
+    hir::{def::DefKind, def_id::DefId},
+    middle::ty::{self, Clause, DefIdTree, ImplSubject, PredicateKind, TyCtxt},
+};
+
+/// data structure to represent the code we want to generate
+/// since we didnt manage to this with syn etc..
```

Comment:
Reword, expand. The reference to `syn` only makes sense to us because we discussed it…